AppinfoJsonFileType.js

```diff
@@ -0,0 +1,168 @@
+/*
+ * AppinfoFileType.js - Represents a collection of appinfo.json files
+ *
+ * Copyright © 2019, JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var fs = require("fs");
+var path = require("path");
+var log4js = require("log4js");
+var AppinfoJsonFile = require("./AppinfoJsonFile.js");
+var logger = log4js.getLogger("loctool.plugin.AppinfoJsonFileType");
+
+var AppinfoJsonFileType = function(project) {
+    this.type = "json";
+    this.resourceType = "json";
+    this.project = project;
+    this.extensions = [".json"];
+
+    /* in order to match proper xliff
+    * datatype value is using to create reskey
+    */
+    var dataTypeMap = {
+        "webos-web":"javascript",
+        "webos-qml": "x-qml",
+        "webos-cpp": "cpp",
+        "webos-c": "cpp"
+    }
+
+    this.datatype = dataTypeMap[project.options.projectType] || "javascript";
+
+    this.API = project.getAPI();
+    this.extracted = this.API.newTranslationSet(project.getSourceLocale());
+    this.newres = this.API.newTranslationSet(project.getSourceLocale());
+    this.pseudo = this.API.newTranslationSet(project.getSourceLocale());
+};
+
+/**
+ * Return true if the given path is a java file and is handled
+ * by the current file type.
+ *
+ * @param {String} pathName path to the file being questions
+ * @returns {boolean} true if the path is a java file, or false
+ * otherwise
+ */
+AppinfoJsonFileType.prototype.handles = function(pathName) {
+    logger.debug("AppinfoJsonFileTyp handles " + pathName + "?");
+    if (!pathName) return false;
+    
+    return (pathName === "appinfo.json") ? true : false
```

Comment:
Should maybe test that the last thing in the path is a file name that is "appinfo.json"

```javascript
path.basename(pathName) === "appinfo.json"
```